experimental/JuLie/partitions.jl

```diff
@@ -0,0 +1,675 @@
+1. "ks" (*default*) is the algorithm "AccelAsc" (Algorithm 4.1) in [KO14](@cite).
+2. "m" is Algorithm 6 in [Mer12](@cite). This is actually similar to "ks".
+
+The ascending partitions are stored here as arrays and are not of type ```Partition``` since the latter are descending by our convention. I am using "ks" as default since it looks slicker and I believe there is a tiny mistake in the publication of "m" (which I fixed).
+
+# Comparison
+
+I don't see a significant speed difference to the descending encoding:
+```julia-repl
+julia> @btime partitions(Int8(90));
+3.376 s (56634200 allocations: 6.24 GiB)
+
+julia> @btime ascending_partitions(Int8(90),alg="ks");
+3.395 s (56634200 allocations: 6.24 GiB)
+
+julia> @btime ascending_partitions(Int8(90),alg="m");
+3.451 s (56634200 allocations: 6.24 GiB)
+```
+"""
+function ascending_partitions(n::Integer; alg="ks")
+
+  #Argument checking
+  n >= 0 || throw(ArgumentError("n >= 0 required"))
+
+  # Use type of n
+  T = typeof(n)
+
+  # Some trivial cases
+  if n==0
+    return Vector{T}[ [] ]
+  elseif n==1
+    return Vector{T}[ [1] ]
+  end
+
+  # Now, the algorithm starts
+  if alg=="ks"
+    P = Vector{T}[]    #this will be the array of all partitions
+    a = zeros(T, n)
+    k = 2
+    y = n-1
+    while k != 1
+      k -= 1
+      x = a[k] + 1
+      while 2*x <= y
+        a[k] = x
+        y -= x
+        k += 1
+      end
+      l = k + 1
+      while x <= y
+        a[k] = x
+        a[l] = y
+        push!(P, a[1:l])
+        x += 1
+        y -= 1
+      end
+      y += x - 1
+      a[k] = y + 1
+      push!(P, a[1:k])
+    end
+    return P
+
+  elseif alg=="m"
+    P = Vector{T}[]    #this will be the array of all partitions
+    a = zeros(T, n)
+    k = 1
+    x = 1
+    y = n-1
+    while true
+      while 3*x <= y
+        a[k] = x
+        y = y-x
+        k += 1
+      end
+      t = k + 1
+      u = k + 2
+      while 2*x <= y
+        a[k] = x
+        a[t] = x
+        a[u] = y - x
+        push!(P, a[1:u])
+        p = x + 1
+        q = y - p
+        while p <= q
+          a[t] = p
+          a[u] = q
+          push!(P, a[1:u])
+          p += 1
+          q -= 1
+        end
+        a[t] = y
+        push!(P, a[1:t])
+        x += 1
+        y -= 1
+      end
+      while x<=y
+        a[k] = x
+        a[t] = y
+        push!(P, a[1:t])
+        x += 1
+        y -= 1
+      end
+      y += x-1
+      a[k] = y+1
+      push!(P, a[1:k])
+      k -= 1
+
+      # I think there's a mistake in the publication
+      # because here k could be zero and then we access
+      # a[k].
+      # That's why I do a while true and check k > 0 here.
+      if k == 0
+        break
+      else
+        x = a[k] + 1
+      end
+    end
+    return P
```

Comment:
So if `alg` is an unsupported value, we just run into an error about P being unassigned here? Perhaps better to add an explicit `error` inside an `else` block above?